src/Adapter/Product/GeneralConfiguration.php

```diff
@@ -65,15 +81,17 @@ public function __construct(
      */
     public function getConfiguration()
     {
+        $shopConstraint = $this->getShopConstraint();
+
         return [
-            'catalog_mode' => $this->configuration->getBoolean('PS_CATALOG_MODE'),
-            'catalog_mode_with_prices' => $this->configuration->getBoolean('PS_CATALOG_MODE_WITH_PRICES'),
-            'new_days_number' => $this->configuration->get('PS_NB_DAYS_NEW_PRODUCT'),
-            'short_description_limit' => $this->configuration->get('PS_PRODUCT_SHORT_DESC_LIMIT'),
-            'quantity_discount' => $this->configuration->get('PS_QTY_DISCOUNT_ON_COMBINATION'),
-            'force_friendly_url' => $this->configuration->getBoolean('PS_FORCE_FRIENDLY_PRODUCT'),
-            'default_status' => $this->configuration->getBoolean('PS_PRODUCT_ACTIVATION_DEFAULT'),
-            'specific_price_priorities' => $this->getPrioritiesData(),
+            'catalog_mode' => (bool) $this->configuration->get('PS_CATALOG_MODE', false, $shopConstraint),
+            'catalog_mode_with_prices' => (bool) $this->configuration->get('PS_CATALOG_MODE_WITH_PRICES', false, $shopConstraint),
+            'new_days_number' => (int) $this->configuration->get('PS_NB_DAYS_NEW_PRODUCT', 0, $shopConstraint),
+            'short_description_limit' => (int) $this->configuration->get('PS_PRODUCT_SHORT_DESC_LIMIT', 0, $shopConstraint),
+            'quantity_discount' => (int) $this->configuration->get('PS_QTY_DISCOUNT_ON_COMBINATION', 0, $shopConstraint),
+            'force_friendly_url' => (bool) $this->configuration->get('PS_FORCE_FRIENDLY_PRODUCT', false, $shopConstraint),
+            'default_status' => (bool) $this->configuration->get('PS_PRODUCT_ACTIVATION_DEFAULT', false, $shopConstraint),
+            'specific_price_priorities' => (array) $this->getPrioritiesData(),
```

Comment:
```suggestion
            'specific_price_priorities' => $this->getPrioritiesData(),
```

Strict typed as an array for the return value.